File: app/src/main/java/me/imzack/app/ender/view/contract/PlanSearchViewContract.java
package me.imzack.app.ender.view.contract;

import me.imzack.app.ender.view.adapter.PlanSearchListAdapter;

public interface PlanSearchViewContract extends BaseViewContract {

    void showInitialView(int planCount, PlanSearchListAdapter planSearchListAdapter);

    void onSearchChanged(boolean isNoSearchInput, boolean isPlanSearchEmpty);

    void onPlanItemClicked(int planListPos);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/PlanSearchViewContract.kt
package net.zackzhang.app.end.view.contract

import net.zackzhang.app.end.view.adapter.PlanSearchListAdapter

interface PlanSearchViewContract : BaseViewContract {

    fun showInitialView(planCount: Int, planSearchListAdapter: PlanSearchListAdapter)

    fun onSearchChanged(isNoSearchInput: Boolean, isPlanSearchEmpty: Boolean)

    fun onPlanItemClicked(planListPos: Int)
}
